drivers/i2c/busses/i2c-imx.c

```diff
@@ -20,6 +20,8 @@
  *
```

Comment:
Since there is no way to comment on the commit log, I'll comment here.

As far as I know, the RF link up indication is accessible from the modem control API that communicates with the K60 MCU over UART. The I2C interface does not provide link up indication. I2C only provides information about the link quality, modulation profile, and transfer rates. This allows the host software running on the i.MX6 SoC to decide on the best modulation.

Reply:
host/app/modem/modem_nxp/modem/statistics.cpp contains the current state machine for determining link status depending on I2C events that the i.MX6 reads from the k60.

The "new BSP" effort was defined as a porting effort only, not a rewrite of the PHY/modem code. The rewrite of the PHY/modem code task was assigned to you AFAIK.